Assets/HoloToolkit/UX/Scripts/Buttons/Button.cs

```diff
@@ -0,0 +1,488 @@
+        /// FocusManager SendMessage("FocusEnter") receiver.
+        /// </summary>
+        public void OnFocusEnter(PointerSpecificEventData eventData)
+        {
+            if (!m_disabled)
+            {
+                ButtonStateEnum newState = m_bHandVisible ? ButtonStateEnum.Targeted : ButtonStateEnum.ObservationTargeted;
+                this.OnStateChange(newState);
+
+                m_bFocused = true;
+            }
+        }
+
+        /// <summary>
+        /// FocusManager SendMessage("FocusExit") receiver.
+        /// </summary>
+        public void OnFocusExit(PointerSpecificEventData eventData)
+        {
+             if (!m_disabled) // && FocusManager.Instance.IsFocused(this))
+            {
+                if (ButtonState == ButtonStateEnum.Pressed)
+                {
+                    DoButtonCancelled();
+                }
+
+                ButtonStateEnum newState = m_bHandVisible ? ButtonStateEnum.Interactive : ButtonStateEnum.Observation;
+
+                if (RequireGaze || ButtonState != ButtonStateEnum.Pressed)
+                {
+                    this.OnStateChange(newState);
+                }
+
+                m_bFocused = false;
+            }
+        }
+
+        /// <summary>
+        /// On Source detected see if it is a hand and increment hand count and set visiblity
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnSourceDetected(SourceStateEventData eventData)
+        {
+            InteractionSourceInfo sourceInfo;
+            if (eventData.InputSource.TryGetSourceKind(eventData.SourceId, out sourceInfo))
+            {
+                if (sourceInfo == InteractionSourceInfo.Hand)
+                {
+                    m_HandCount++;
+                    m_bHandVisible = true;
+                }
+            }
+        }
+
+        /// <summary>
+        ///  On Source lost decrement hand count and set visiblity
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnSourceLost(SourceStateEventData eventData)
+        {
+            InteractionSourceInfo sourceInfo;
+            if (eventData.InputSource.TryGetSourceKind(eventData.SourceId, out sourceInfo))
+            {
+                if (sourceInfo == InteractionSourceInfo.Hand)
+                {
+                    m_HandCount--;
+                    m_bHandVisible = m_HandCount <= 0;
+                }
+            }
+        }
+        #endregion
+
+        /// <summary>
+        /// Called when button is pressed down.
+        /// </summary>
+        protected void DoButtonPressed(bool bRelease = false)
+        {
+            ButtonStateEnum newState = ButtonStateEnum.Pressed;
+            this.OnStateChange(newState);
+
+            if (OnButtonPressed != null)
+            {
+                OnButtonPressed(gameObject);
+            }
+
+            if (bRelease)
+            {
+                StartCoroutine("DelayedRelease", 0.2f);
```

Comment:
nit: use `IEnumerator` call to `StartCoroutine` so we can get symbol completion: `StartCoroutine(DelayedRelease(0.2f));`